module/signature/errors.go

```diff
@@ -5,21 +5,92 @@ import (
 	"fmt"
 )
 
-var (
-	ErrInvalidFormat      = errors.New("invalid signature")
-	ErrInsufficientShares = errors.New("insufficient threshold signature shares")
-)
+var ErrInsufficientShares = errors.New("insufficient threshold signature shares")
+
+/* ********************* InvalidSignatureIncludedError ********************* */
+
+// InvalidSignatureIncludedError indicates that some signatures, included via TrustedAdd, are invalid
+type InvalidSignatureIncludedError struct {
+	err error
+}
+
+func NewInvalidSignatureIncludedErrorf(msg string, args ...interface{}) error {
+	return InvalidSignatureIncludedError{
+		err: fmt.Errorf(msg, args...),
+	}
+}
+
+func (e InvalidSignatureIncludedError) Error() string { return e.err.Error() }
+func (e InvalidSignatureIncludedError) Unwrap() error { return e.err }
 
-// ErrDuplicatedSigner is returned when an API detects a duplicate signer when it shouldn't
-type ErrDuplicatedSigner struct {
-	message string
+// IsInvalidSignatureIncludedError returns whether err is an InvalidSignatureIncludedError
+func IsInvalidSignatureIncludedError(err error) bool {
+	var e InvalidSignatureIncludedError
+	return errors.As(err, &e)
 }
 
-// newErrDuplicatedSigner constructs a new ErrDuplicatedSigner
-func newErrDuplicatedSigner(msg string, args ...interface{}) error {
-	return &ErrDuplicatedSigner{message: fmt.Sprintf(msg, args...)}
+/* ************************* InvalidSignerIdxError ************************* */
+
+// InvalidSignerIdxError indicates that the signer index is invalid
+type InvalidSignerIdxError struct {
+	err error
```

Comment:
Same comment for the embedded error and deleting `Error()` , although this is very minor and there isn't a need to update

Reply:
see my earlier comment: https://github.com/onflow/flow-go/pull/1830#discussion_r787279755